### PR TITLE
NFDIV-770

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-770

### CODE REVIEW

File: src/main/steps/applicant1/application-ended/get.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  constructor() {
+    super(__dirname + '/template', generateContent);
+  }
+
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    const caseworkerUser = await getSystemUser();
+    req.locals.api = getCaseApi(caseworkerUser, req.locals.logger);
+
+    try {
+      req.session.userCase.applicationType = ApplicationType.SOLE_APPLICATION;
```
Comment (velimd) [inline src/main/steps/applicant1/application-ended/get.ts:24]: i feel like this whole logic should be in the post controller for the previous page.
user clicks switch to sole -> trigger event in post -> redirect to application ended.

Comment (tomxelliott) [reply src/main/steps/applicant1/application-ended/get.ts:24]: The flow that leads the user to this page is the log in page that is handled by Idam. The user logs in after app2 has said No the marriage hasn't broken down. Therefore there is no post on a page prior to this - only the log in page. If there was was a page before then, yes I would have done that.

File: src/main/steps/applicant1/application-ended/get.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import * as caseApi from '../../../app/case/CaseApi';
+import { generatePageContent } from '../../common/common.content';
+
+import { generateContent } from './content';
+import ApplicationEndedGetController from './get';
+
+const getSystemUserMock = jest.spyOn(oidc, 'getSystemUser');
+const getCaseApiMock = jest.spyOn(caseApi, 'getCaseApi');
+
+describe('AccessCodeGetController', () => {
```
Comment (lara-mpt) [inline src/main/steps/applicant1/application-ended/get.test.ts:14]:
```suggestion
describe('ApplicationEndedGetController', () => {
```

File: src/main/steps/applicant1/application-ended/get.ts (hunk 1 above)
Comment (velimd) [reply src/main/steps/applicant1/application-ended/get.ts:24]: fair

**Review by velimd — APPROVED** [review]

**Review by lara-mpt — APPROVED** [review]

**Pull request merged**